fdbrpc/DDSketchTest.actor.cpp

```diff
@@ -0,0 +1,62 @@
+#include "fdbrpc/DDSketch.h"
+#include "flow/Error.h"
+#include "flow/IRandom.h"
+#include "flow/UnitTest.h"
+#include <limits>
+#include <random>
+#include "flow/actorcompiler.h" // has to be last include
+void forceLinkDDSketchTests() {}
+
+TEST_CASE("/fdbrpc/ddsketch/accuracy") {
+
+	int TRY = 100, SIZE = 1e6;
+	const int totalPercentiles = 7;
+	double targetPercentiles[totalPercentiles] = { .0001, .01, .1, .50, .90, .99, .9999 };
+	double stat[totalPercentiles] = { 0 };
+	for (int t = 0; t < TRY; t++) {
+		DDSketch<double> dd;
+		std::vector<double> nums;
+		for (int i = 0; i < SIZE; i++) {
+			static double a = 1, b = 1; // a skewed distribution
+			auto y = deterministicRandom()->random01();
+			auto num = b / pow(1 - y, 1 / a);
+			nums.push_back(num);
+			dd.addSample(num);
+		}
+		std::sort(nums.begin(), nums.end());
+		for (int percentID = 0; percentID < totalPercentiles; percentID++) {
+			double percentile = targetPercentiles[percentID];
+			double ground = nums[percentile * (SIZE - 1)], ddvalue = dd.percentile(percentile);
+			double relativeError = fabs(ground - ddvalue) / ground;
+			stat[percentID] += relativeError;
+		}
+	}
+
+	for (int percentID = 0; percentID < totalPercentiles; percentID++) {
+		printf("%.4lf per, relative error %.4lf\n", targetPercentiles[percentID], stat[percentID] / TRY);
+	}
+
+	return Void();
+}
+
+TEST_CASE("/fdbrpc/ddsketch/correctness") {
+	DDSketch<double> dd;
+
+	for (int i = 0; i < 4000; i++) {
+		// This generates a uniform real disitribution between the range of
```

Comment:
"uniform distribution" seems inconsistent with "randomSkewedUint32"

Reply:
The reason why I said "uniform distribution" is because `randomSkewedUint32` used that internally (along with other things). I really just want something that generates small numbers with a decent amount of spread (and is deterministic of course). 

I was going to use ` std::uniform_real_distribution` directly but couldn't find a way to get the simulation seed to populate it.

Reply:
Usually folks apply a linear function to random01() to get the distribution they want

Reply:
Good idea